src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Microsoft.Android.Sdk.Bindings.Gradle.targets

```diff
@@ -0,0 +1,93 @@
+    </AndroidGradleProjectReference>
+  </ItemDefinitionGroup>
+
+
+  <Target Name="_GetBuildAndroidGradleProjectsInputs">
+    <ItemGroup>
+      <_GradleInputs Include="$(MSBuildProjectFullPath)" />
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.java" />
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.kt" />
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.kts" />
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.gradle" />
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.properties"/>
+      <_GradleInputs Include="%(AndroidGradleProjectReference.FullPath)/**/*.xml" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="BuildAndroidGradleProjects"
+      Condition=" '@(AndroidGradleProjectReference->Count())' != '0' "
+      DependsOnTargets="_GetBuildAndroidGradleProjectsInputs"
+      Inputs="@(_GradleInputs)"
+      Outputs="$(_BuildAndroidGradleProjectsStamp)" >
+    <!-- Delete outputs folder if inputs have changed to force a partial rebuild -->
+    <RemoveDir Directories="@(AndroidGradleProjectReference->'%(OutputPath)outputs')" />
+
+    <!-- Run assemble task for project outputs, android app and library project outputs are currently supported -->
+    <Gradle ToolPath="%(AndroidGradleProjectReference.FullPath)"
+        Command="assemble%(AndroidGradleProjectReference.Configuration)"
+        ModuleName="%(AndroidGradleProjectReference.ModuleName)"
+        OutputPath="%(AndroidGradleProjectReference.OutputPath)"
+        IntermediateOutputPath="$(IntermediateOutputPath)"
+        AndroidSdkDirectory="$(AndroidSdkDirectory)"
+        JavaSdkDirectory="$(JavaSdkDirectory)" >
+      <Output TaskParameter="AppOutputs" ItemName="AndroidGradleProjectAppOutputs" />
+      <Output TaskParameter="LibraryOutputs" ItemName="AndroidGradleProjectLibraryOutputs" />
+    </Gradle>
+
+    <Touch Files="$(_BuildAndroidGradleProjectsStamp)" AlwaysCreate="True" />
+    <ItemGroup>
+      <FileWrites Include="$(_BuildAndroidGradleProjectsStamp)" />
+    </ItemGroup>
+
+    <!-- Automatically add library outputs to be bound or otherwise processed later in the build -->
+    <ItemGroup>
+      <AndroidLibrary Include="@(AndroidGradleProjectLibraryOutputs)" Condition=" '%(AndroidGradleProjectReference.ReferenceLibraryOutputs)' == 'true' ">
+        <Bind>%(AndroidGradleProjectReference.Bind)</Bind>
+        <Pack>%(AndroidGradleProjectReference.Pack)</Pack>
+        <Visible>%(AndroidGradleProjectReference.Visible)</Visible>
+      </AndroidLibrary>
+    </ItemGroup>
+
+    <Message Text="Adding reference to Gradle output: @(AndroidGradleProjectLibraryOutputs). The %25(ReferenceLibraryOutputs) metadata can be set to false to opt out of this behavior."
+        Condition=" '@(AndroidGradleProjectLibraryOutputs->Count())' != '0' and '%(AndroidGradleProjectReference.ReferenceLibraryOutputs)' == 'true' " />
```

Comment:
Do we need to localize this message, or is a debug message?

Reply:
I think I'd want to localize this, I just added a resource for it.